packages/picasso/src/CalendarDay/CalendarDay.tsx

```diff
@@ -0,0 +1,142 @@
+import CalendarContext from '../CalendarContext'
+
+interface RenderDayProps extends DayProps {
+  /** Unique key */
+  key?: string
+  /** Children nodes */
+  children?: ReactNode
+  /** Returns formatted date */
+  getDayFormatted: (date: Date) => string
+  /** Specifies if current day belongs to the previous month */
+  isMonthPrev: boolean
+  /** Specifies if current day belongs to the next month */
+  isMonthNext: boolean
+  /** Specifies if day can be selected */
+  isSelectable: boolean
+}
+
+export type RenderDay = (args: RenderDayProps) => ReactNode
+
+/** Represent the props used by the {@link Day} component. */
+export interface CalendarDayProps extends DayProps {
+  /** Unique key */
+  key?: string
+}
+
+const getDayFormatted = (date: Date) => format(date, 'd')
+
+const checkIfBelongsToPreviousMonth = (date: Date, currentMonth: Date) =>
+  isBefore(date, startOfMonth(currentMonth))
+
+const checkIfBelongsToNextMonth = (date: Date, currentMonth: Date) =>
+  isAfter(date, endOfMonth(currentMonth))
+
+const useStyles = makeStyles<Theme>(styles, { name: 'PicassoCalendar' })
+
+/**
+ * The content of a day cell
+ */
+const CalendarDay = (dayProps: CalendarDayProps): JSX.Element => {
+  const { key, date, displayMonth } = dayProps
+  const buttonRef = useRef<HTMLButtonElement>(null)
+  const { onDayMouseEnter, renderDay } = useContext(CalendarContext)
+
+  const { activeModifiers, buttonProps } = useDayRender(
+    date,
+    displayMonth,
+    buttonRef
+  )
+  const {
+    selected: isSelected,
+    disabled: isDisabled,
+    indicated: isIndicated,
+    outside: isOutside,
+    weekend: isWeekend,
+    temporaryRangeMiddle: isTemporaryRangeMiddle,
+    temporaryRangeEnd: isTemporaryRangeEnd,
+    range_start: isRangeStart,
+    range_middle: isRangeMiddle,
+    range_end: isRangeEnd,
+  } = activeModifiers
+
+  const classes = useStyles()
+
+  const defaultMarkup = (
+    <button
+      ref={buttonRef}
+      type='button'
+      data-testid={`day-button-${
+        isSelected ? 'selected' : getDayFormatted(date)
+      }`}
+      key={key}
+      tabIndex={isDisabled ? -1 : undefined}
+      className={cx(classes.day, {
+        [classes.selected]: isSelected,
+        [classes.weekend]: isWeekend,
+        [classes.grayed]: isOutside && !isDisabled,
+        [classes.disabled]: isDisabled,
+        [classes.startSelection]: isRangeStart,
+        [classes.withinSelection]: isRangeMiddle || isTemporaryRangeMiddle,
+        [classes.endSelection]: isRangeEnd || isTemporaryRangeEnd,
+      })}
+      onClick={
+        isDisabled
+          ? undefined
+          : event => {
+              if (buttonProps?.onClick) {
+                buttonProps.onClick(event)
+              }
+            }
+      }
+      onMouseEnter={event => {
+        if (onDayMouseEnter) {
+          onDayMouseEnter(date)
+        }
+
+        if (buttonProps.onMouseEnter) {
+          buttonProps.onMouseEnter(event)
+        }
```

Comment:
NIT: We can also simplify here
```suggestion
        onDayMouseEnter?.(date)
        buttonProps.onMouseEnter?.(event)
```

Reply:
@augustobmoura Thank you for the suggestion, changed!